### PR TITLE
Improve handling of invalid topic configurations

### PR DESCRIPTION
**Pull request by fvaleri**
### Type of change

Improvement

### Description

An invalid topic configuration results in a RuntimeException that fails the whole reconciliation and following periodic reconciliations. Additionally, if you have a batch of 100 topics or more, it can be difficult to spot where the configuration errors is.

With this change the TO reports the error in the status of the Kafka topic containing the invalid configuration, and continues with the reconciliation.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

File: topic-operator/src/main/java/io/strimzi/operator/topic/BatchingTopicController.java [hunk 1]
```diff
@@ -292,10 +292,15 @@ private static Either<TopicOperatorException, Boolean> validateUnchangedTopicNam
     }
 
     private PartitionedByError<ReconcilableTopic, Void> createTopics(List<ReconcilableTopic> kts) {
+        Map<ReconcilableTopic, TopicOperatorException> newTopicsErrors = new HashMap<>();
         var newTopics = kts.stream().map(reconcilableTopic -> {
-            // Admin create
```
Comment (ShubhamRwt) [inline topic-operator/src/main/java/io/strimzi/operator/topic/BatchingTopicController.java:296]: Was this comment relevant?

Comment (fvaleri) [reply topic-operator/src/main/java/io/strimzi/operator/topic/BatchingTopicController.java:296]: I think it is a leftover, as the admin call is few lines below with a clear log line.

**Comment by tombentley** [comment]
@fvaleri I pushed a few tweaks to your branch. Could you take a look and let me know if you agree with them, if not we can discuss here. Thanks.

**Comment by fvaleri** [comment]
Thanks for the improvement. It looks good, I just did a rebase due to an import conflict. We are good to go now.

**Review by tombentley — APPROVED** [review]
LGTM, thanks @fvaleri

**Pull request merged**